### PR TITLE
Add mapper using kotlinx serialization

### PR DESCRIPTION
**Pull request by MrNuggelz**
This pull request adds a new mapper that uses [kotlinx.serialization](https://github.com/Kotlin/kotlinx.serialization) instead of jackson.

### CODE REVIEW

**Comment by anidotnet** [comment]
Hi @MrNuggelz thanks a lot for your contribution. Unfortunately, just today I have pushed some significant changes in Nitrite's main branch. Like - build system has been changed to maven, `NitriteMapper` api has been changed. 

Could you please raise this PR against the main branch and change your code according to the recent `NitriteMapper` change? I am extremely sorry for the inconvenience.

**Comment by anidotnet** [comment]
And while you are implementing kotlinx serialization based `NitriteMapper` would you be so kind to check this issue #815, if you are solution can address the issue mentioned there?

File: potassium-nitrite/pom.xml [hunk 1]
```diff
@@ -104,6 +104,11 @@
             <artifactId>snakeyaml</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jetbrains.kotlinx</groupId>
+            <artifactId>kotlinx-serialization-json</artifactId>
+            <version>1.5.1</version>
```
Comment (anidotnet) [inline potassium-nitrite/pom.xml:110]: Could you please move this version declaration in parent pom?

File: potassium-nitrite/pom.xml [hunk 2]
```diff
@@ -114,6 +119,27 @@
                 <groupId>org.jetbrains.kotlin</groupId>
                 <artifactId>kotlin-maven-plugin</artifactId>
                 <extensions>true</extensions>
+                <executions>
```
Comment (anidotnet) [inline potassium-nitrite/pom.xml:122]: Please move all plugin config in pluginManagement in parent pom.xml

**Review by anidotnet — CHANGES_REQUESTED** [review]
Could you please make the changes?

**Comment by MrNuggelz** [comment]
Hi [anidotnet](https://github.com/anidotnet). Thanks for the fast review.
I looked into #815 and what would need to be changed. I think the `tryConvert` method from `NitriteMapper` would require an additional parameter for the source class `Class<Source> sourceType`. I am currently trying to implement this.
Also is there a way to ignore some of the lining error from the "Codacy Static Code Analysis"? All of the remaining errors are can't do not make sense/really be fixed.

**Comment by anidotnet** [comment]
About the Codacy Static Code Analysis. I have addressed high level issues earlier. All Medium and Minor issues has to be addressed based on the context and viability. Rest of them should be ignored. When I get some time, I'll get into it.

**Pull request merged**